corehq/apps/sso/tests/test_tasks.py

```diff
@@ -133,3 +139,103 @@ def tearDown(self):
     def tearDownClass(cls):
         cls.account.delete()
         super().tearDownClass()
+
+
+class TestAutoDeactivationTask(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.account = generator.get_billing_account_for_idp()
+        cls.account.enterprise_admin_emails = ['test@vaultwax.com']
+        cls.account.save()
+        cls.domain = Domain.get_or_create_with_name("vaultwax-001", is_active=True)
+
+        enterprise_plan = accounting_generator.subscribable_plan_version(edition=SoftwarePlanEdition.ENTERPRISE)
+        accounting_generator.generate_domain_subscription(
+            cls.account,
+            cls.domain,
+            date_start=datetime.date.today(),
+            date_end=None,
+            plan_version=enterprise_plan,
+            is_active=True,
+        )
+
+        cls.idp = generator.create_idp('vaultwax', cls.account)
+        cls.idp.enable_user_deactivation = True
+        cls.idp.idp_type = IdentityProviderType.AZURE_AD
+        cls.idp.save()
+        cls.email_domain = AuthenticatedEmailDomain.objects.create(
+            email_domain='vaultwax.com',
+            identity_provider=cls.idp,
+        )
+
+    def setUp(self):
+        super().setUp()
+        self.web_user_a = self._create_web_user('a@vaultwax.com')
+        self.web_user_b = self._create_web_user('b@vaultwax.com')
+        self.web_user_c = self._create_web_user('c@vaultwax.com')
+        self.web_user_d = self._create_web_user('d@vaultwax.com')
```

Comment:
Is this user ever needed for any of these tests?

Reply:
It is needed because of `len(web_user_in_account) - len(exempt_usernames) > 3`. I can left a comment here: 28aaa49c3bf080454d2f36941b64d7a2f7d129e9

Reply:
I see. Doesn't require any action, but the downside of setting up these users in setUp is that you can't easily test the scenario where auto-deactivation is _not_ skipped because  `len(web_user_in_account)_ - len(exempt_usernames) <= 3`. Is that fair to say?

Reply:
Yes that's fair. :face_holding_back_tears: